pkg/reconciler/instances/istio/action.go

```diff
@@ -23,20 +40,174 @@ func (a *ReconcileAction) Run(version, profile string, config map[string]interfa
 		return err
 	}
 
-	performer, err := actions.NewDefaultIstioPerformer(context.KubeClient.Kubeconfig(), manifest.Manifest, context.KubeClient, context.Logger, &istioctl.DefaultCommander{})
+	ver, err := a.performer.Version(context.WorkspaceFactory, version, istioChart, context.KubeClient.Kubeconfig(), context.Logger)
```

Comment:
I would propose to rename "Version" to something more meaningful like for e.g. "GetInstalledIstioVersion"